README.rst

```diff
@@ -1241,6 +1241,36 @@ like :code:`<session>.permissions`, except that the :code:`<session>.acls.get(..
 :code:`<session>.permissions` is therefore deprecated and, in v2.0.0, will be removed in favor of :code:`<session>.acls`.
 
 
+Quotas (v1.1.9+)
+----------------
+
+Quotas may be set for a group:
+
+    session.groups.set_quota('quota_group', 100000, resource = 'my_limited_resource')
+
+or per user, before iRODS 4.2.12:
+
+    session.users.set_quota('alice', 100000)
+
+(The default for the resource parameter is "total", denoting a general quota usage not bound to a particular resource.)
+
+The Quota model is also available for queries.  So, to determine the space remaining for a certain group on a given resource:
+
+    from irods.models import Quota
+    session.groups.calculate_usage()
+    group, resource = ['my_group','my_resc']
+    space_left_in_bytes = list(session.query(Quota.over).filter(Quota.user_id == session.groups.get(group).id,
+                                                                Quota.resc_id == session.resources.get(resource).id))[0][Quota.over] * -1
+
+And, to remove all quotas for a given group, one might (as a rodsadmin) do the following:
+
+    from irods.models import Resource, Quota
+    resc_map = dict([(x[Resource.id],x[Resource.name]) for x in sess.query(Resource)] + [(0,'total')])
+    group = sess.groups.get('mygroup')
+    for quota in sess.query(Quota).filter(Quota.user_id == group.id):
+        sess.groups.remove_quota(group.name,  resource = resc_map[quota.resc_id])
```

Comment:
Is `[(0, 'total')]` a reachable case? `quota.resc_id` should never return 0.

Reply:
It's a cheat code .... We're taking advantage of the fact  that the `Quota.resc_id` is returned as 0 for the case of "total" quotas.  This way, the parameter will be sent as "total", and after the loop is done, all quotas for that group are guaranteed to be cleared away.  (Barring race conditions, of course.)

Reply:
I see. So, querying info about quotas will result in a resc id of 0 when the quota is for the "total". I'm guessing that's a feature of the PRC and not something iRODS is doing?

Reply:
That's an edge case of what the GenQuery returns when there is no resource assigned for the quota in question.
For example:
```
$ iadmin  sgq  public total 100000
$ iadmin lu public|grep user_id
user_id: 10001
$ iquest "select QUOTA_RESC_ID where QUOTA_USER_ID = '10001'"
QUOTA_RESC_ID = 0
```

Reply:
Got it. It's a result of how GenQuery handles queries about quotas.

It makes sense.

Carry on.